modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/EditFileEntryMVCActionCommand.java

```diff
@@ -1400,7 +1401,7 @@ else if (cmd.equals(Constants.ADD_DYNAMIC)) {
 	}
 
 	private void _validateFileName(String sourceFileName, String extension)
-		throws FileNameExtensionException {
+		throws FileNameExtensionException, FileNameExtensionMismatchException {
```

Comment:
Can you make the method declare only `PortalException`? When there are multiple exceptions with a common superclass, the superclass has to be used instead.